src/worker_pool/base.ts

```diff
@@ -2,6 +2,7 @@ import assert from 'node:assert';
 
 export abstract class AsynchronouslyCreatedResource {
     onreadyListeners : (() => void)[] | null = [];
+    ondestroyListeners : (() => void)[] | null = [];
```

Comment:
Why not make `AsynchronouslyCreatedResource` extends `EventEmitter`?

Reply:
So we emit events with the updated state of the worker?

Reply:
The existing code seems to re-eimplement `EventEmitter` semantic. I do not know the rationale behind it.

Reply:
See what you mean; me neither, I imagine the idea was to implement a lightweight version of the EventEmitter; let me see if there's some potential performance overhead that was tried to be avoided, otherwise can switch to it.

Reply:
After replacing, and using an `EventEmitter` instead, the performance dropped at least 30%. Then it confirms it was mostly due to small use case and to reduce overhead